addons/mail/models/mail_rotting_stage_mixin.py

```diff
@@ -0,0 +1,10 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class RottingStageMixin(models.AbstractModel):
```

Comment:
Would not to a mixin for that. Instead would define an helper `_mail_get_track_stage_field`` that returns the "stage_id field" (just an helper around ``_track_duration_field`` defined in duration mixin, that we keep for compatibility). Then `_mail_get_track_stage_day_rot_field` returning the rot field on the stage (default being 'day_rot'). If not defined and used for rotting feature, raise an UserError when calling it.